frontend/catalyst/api_extensions/quantum_operators.py

```diff
@@ -392,16 +394,28 @@ class MidCircuitMeasure(HybridOp):
 
     binder = qmeasure_p.bind
 
+    def __init__(self, *args, **kwargs):
+        self.bypass_postselect = kwargs.pop("bypass_postselect", False)
+        self.postselect = kwargs.pop("postselect", None)
+        self.reset = kwargs.pop("reset", False)
+        super().__init__(*args, **kwargs)
+
     def trace_quantum(self, ctx, device, trace, qrp) -> QRegPromise:
         op = self
         wire = op.in_classical_tracers[0]
         qubit = qrp.extract([wire])[0]
-        postselect = op.in_classical_tracers[1]
-
-        qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit, postselect=postselect)
+        if self.bypass_postselect:
+            qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit)
```

Comment:
Ah, saw this property in your other PR. If the transform wants to turn a post-selected measurement into a regular one, wouldn't it be cleaner to simply set `postselect` to `None`?

Edit: Nvm, I guess you still need to keep track of the fact that you are post-selecting for the post-processing right?